funnel/forms/proposal.py

```diff
@@ -74,10 +74,15 @@ def __init__(self, *args, **kwargs):
             self.description.description = space.proposal_part_b.get('hint')
 
 
-class ProposalStatusForm(forms.Form):
-    status = forms.SelectField(
-        __("Status"), coerce=int,
-        choices=[(status, label.title) for (status, label) in PROPOSALSTATUS.items() if status != PROPOSALSTATUS.DRAFT])
+class ProposalTransitionForm(forms.Form):
+    transition = forms.SelectField(__("Status"), validators=[forms.validators.DataRequired()])
+
+    def populate_transitions(self, proposal):
+        # value: transition method name
+        # label: method name in title case, after removing underscores
+        self.transition.choices = sorted([
+            (tname, tname.replace("_", " ").title()) for (tname, t) in proposal.state.transitions.iteritems() if t.is_available
+        ])
```

Comment:
Transitions are actions, not choices. A `SelectField` is the wrong UI element. Remove this form entirely and create a page with buttons. Process available transitions in the view.